plugins/Installation/ServerFilesGenerator.php

```diff
@@ -172,20 +172,26 @@ protected static function createWebConfigFiles()
 </configuration>'
         );
 
+        $additionForPlugins = '
+        <alwaysAllowedUrls>
+          <add url="/plugins/HeatmapSessionRecording/configs.php" />
+        </alwaysAllowedUrls>';
+
+        $additionForMisc = '
+        <alwaysAllowedUrls>
+          <add url="/misc/cron/archive.php" />
+        </alwaysAllowedUrls>';
+
         // deny direct access to .php files
         $directoriesToProtect = array(
             '/libs',
             '/vendor',
-            '/plugins',
+            '/plugins' => $additionForPlugins,
             '/node_modules',
+            '/misc' => $additionForMisc,
         );
 
-        $additionForPlugins = '
-        <alwaysAllowedUrls>
-          <add url="/plugins/HeatmapSessionRecording/configs.php" />
-        </alwaysAllowedUrls>';
-
-        foreach ($directoriesToProtect as $directoryToProtect) {
+        foreach ($directoriesToProtect as $directoryToProtect => $additions) {
             @file_put_contents(
                 PIWIK_INCLUDE_PATH . $directoryToProtect . '/web.config',
```

Comment:
Isn't `$directoryToProtect` here an integer for most cases? 🧐

Reply:
Well spotted! I've changed it so every entry in the array is a folder as the key and any additions as the value, so there's no confusion now. Thanks!